gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveUtils.java

```diff
@@ -171,17 +173,34 @@ public static boolean isPartitioned(Table table) {
   }
 
   /**
+   * First check if the user path is exactly the same as the existing path, if so then just return true
+   * Otherwise there could be a fs mismatch, resolve this through the filesystem
+   * If the paths do not exist, then recurse up the parent directories until there is a match, and compare the children
    * @param fs User configured filesystem of the target table
    * @param userSpecifiedPath user specified path of the copy table location or partition
    * @param existingTablePath path of an already registered Hive table or partition
    * @return true if the filesystem resolves them to be equivalent, false otherwise
    */
   public static boolean areTablePathsEquivalent(FileSystem fs, Path userSpecifiedPath, Path existingTablePath) throws IOException {
+    if (userSpecifiedPath == null || existingTablePath == null) {
+      log.error("User path or existing hive table path is null");
+      return false;
+    }
+    if (userSpecifiedPath.toString().equals(existingTablePath.toString())) {
+      return true;
+    }
     try {
       return fs.resolvePath(existingTablePath).equals(fs.resolvePath(userSpecifiedPath));
```

Comment:
Can we add comments here to explain why we want to compare resolve path instead of just comparing the path? Can we compare path without schema here?

Reply:
Would there be any case we can run into an issue where we do ignore the schema? I guess we'd be under the assumption that the user specified fs.uri is at least consistent with the hive provided fs.uri, that there isn't any bad actors that can provide a poorly formed fs.uri

Reply:
I think my confusion here is why we want to recursively check parent path when the existing table path does not exist, that's a bad table in this case right? Shouldn't we fail it?

Reply:
We see a number of tables like this and supposedly they could be cleaned up by some other job, empty tables that are registered are still valid technically from Hive's POV.